pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -129,7 +128,7 @@ func (s *DiscoveryServer) Register(rpcs *grpc.Server) {
 // ( will be removed after change detection is implemented, to double check all changes are
 // captured)
 func (s *DiscoveryServer) periodicRefresh() {
-	envOverride := os.Getenv("V2_REFRESH")
+	envOverride := pilot.RefreshDuration
```

Comment:
Shouldn’t this be a function as it gets evaluated in every call

Reply:
Ic, you mean by calling function we can dynamically update this env value.

Reply:
After another look, I find this func `periodicRefresh` is called only once. Maybe we can do env reload as a follow up .